server/api/signup.js

```diff
@@ -79,33 +79,24 @@ function signup(req, res, next) {
  * @param {Buffer} hashedPassword - The hashed password of the new User. This (hopefully) should be generated by `Crypto.pbkdf2()`.
  * @param {String} salt - The salt used in hashing the password.
  * @param {Callback} callback - A callback which handles what happens after inserting the user.
- *                   Should take in two arguments: an error `error` and the `user` object generated from `insertUser`.
- *                   `user` will be an empty object if an error is thrown, and `error` will be null if there is no error.
+ *                   Should take in one argument: an error `error`.
+ *                   `error` will be null if there is no error.
  * @returns Nothing.
  */
 async function insertUser(db, username, hashedPassword, salt, callback) {
   await db.run(
     "INSERT INTO User (username, hashed_password, salt) VALUES (?, ?, ?)",
     [username, hashedPassword, salt],
-    function (err) {
-      var user = {};
-
+    (err) => {
       // throw an error if there's an issue
       if (err) {
-        callback(err, user);
+        callback(err);
         return;
       }
 
-      // create a user object, handle it elsewhere
-      // TODO: Perhaps this should also generate a user token at this time?
-      user = {
-        user_id: this.lastID,
-        username: username,
-      };
-
-      callback(err, user);
```

Comment:
note: this caused issues with testing, and it wasn't used anywhere in the endpoint

Reply:
Makes sense.